addons/mail/models/mail_thread.py

```diff
@@ -1644,12 +1644,23 @@ def _message_partner_info_from_emails(self, emails, link_mail=False):
 
     def _message_post_process_attachments(self, attachments, attachment_ids, message_values):
         """ Preprocess attachments for mail_thread.message_post() or mail_mail.create().
+        Purpose is to
+
+          * transfer attachments given by ``attachment_ids`` from the composer to
+            the record (if any);
+          * limit attachments manipulation when being a shared user;
+          * create attachments from ``attachments``. If those are linked to the
+            content (body) through CIDs body is updated accordingly;
+
+        :param list(tuple(str,str), tuple(str,str, dict)) attachments : list of attachment
+            tuples in the form ``(name,content)`` or ``(name,content, info)`` where content
+            is NOT base64 encoded;
+        :param list(id, id) attachment_ids: list of existing attachments to link to this message
```

Comment:
Little strange worded like that, maybe "list(id1, id2, ...)".

Or is that the "official" way to describe the list content?

Reply:
Don't know where I copy pasted that. Could be simplified to list attachment_ids probably.